flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/physical/batch/BatchExecSinkRule.scala

```diff
@@ -45,28 +43,34 @@ class BatchExecSinkRule extends ConverterRule(
     val newTrait = rel.getTraitSet.replace(FlinkConventions.BATCH_PHYSICAL)
     var requiredTraitSet = sinkNode.getInput.getTraitSet.replace(FlinkConventions.BATCH_PHYSICAL)
     sinkNode.sink match {
-      case partitionSink: PartitionableTableSink
-        if partitionSink.getPartitionFieldNames != null &&
-          partitionSink.getPartitionFieldNames.nonEmpty =>
-        val partitionFields = partitionSink.getPartitionFieldNames
-        val partitionIndices = partitionFields
-          .map(partitionSink.getTableSchema.getFieldNames.indexOf(_))
-        // validate
-        partitionIndices.foreach { idx =>
-          if (idx < 0) {
-            throw new TableException(s"Partitionable sink ${sinkNode.sinkName} field " +
-              s"${partitionFields.get(idx)} must be in the schema.")
-          }
+      case partitionSink: PartitionableTableSink =>
```

Comment:
It doesn't feel right for current convert logic.

When output to a partitioned table, I think we should first see whether this table is partitioned, through `CatalogTable.isPartitioned()`. If the table is partitioned, then we have to check whether we have a `PartitionableTableSink` to output data. 

Current logic seems to be just opposite. We first check whether we have partition sink capability, then we check whether the table itself is partitioned.

Reply:
You are right, now, we have changed the roles, catalog table is the master.